package.json

```diff
@@ -235,15 +239,17 @@
     "build-client-if-prod": "node -e \"process.env.CALYPSO_ENV === 'production' && process.exit(1)\" || npm run build-client",
     "build-client-if-desktop": "node -e \"process.env.CALYPSO_ENV === 'desktop' && process.exit(1)\" || npm run build-client",
     "sdk": "node bin/sdk-cli.js",
-    "clean": "npm run -s clean:build && npm run -s clean:devdocs && npm run -s clean:public",
+    "clean": "npm run -s clean:build && npm run -s clean:devdocs && npm run -s clean:public && npm run -s clean:packages",
     "clean:build": "npm run -s rm -- build && npm run -s rm -- server/bundler .json && npm run -s rm -- .babel-cache",
     "clean:devdocs": "npm run -s rm -- server/devdocs/search-index.js && npm run -s rm -- server/devdocs/proptypes-index.json && npm run -s rm -- server/devdocs/components-usage-stats.json",
+    "clean:packages": "lerna run clean",
     "clean:public": "npm run -s rm -- public .css .css.map .js .js.map && npm run -s rm -- public/images .svg && npm run -s rm -- public/sections .css .css.map && npm run -s rm -- public/sections-rtl .css .css.map",
     "distclean": "npm run -s clean && npm run -s rm -- node_modules",
     "docker": "docker run -it --name wp-calypso --rm -p 80:3000 wp-calypso",
     "eslint-branch": "node bin/eslint-branch.js",
     "install-if-deps-outdated": "node bin/install-if-deps-outdated.js",
     "install-if-no-packages": "node bin/install-if-no-packages.js",
+    "postinstall": "lerna bootstrap",
```

Comment:
If the root `package.json` contains `file:*` links, this line is infinity-loopy. `lerna boostrap` runs `npm install` and that triggers the `postinstall` script... ∞

Reply:
Right, if we use `file:` links then we don't run `lerna bootstrap` at all. We also lose the ability to specify loose semver ranges for packages that depend on one another. That's not really a problem yet, so I'm tempted to punt on it, but also want to set the right foundation and not have to tear it up later. 

The problem is: 
* assume we have three packages in packages/*, A, B, C, all at 1.0.0. A and B both depend on C. 
* We publish C@2.0.0. A can use either 1 or 2, but B has to use 2. So we would want A to have a semver dep of `^1 || ^2` and B to have `^2`. But with file refs, the version is always changed to `^$current`, so `^2.0.0`

A related problem is that every update is `^$current`, so even if your package supports `^2.0.0`, you'll get `^2.3.4`.

Using semver refs forces you to think about these issues and declare what you're compatible with. It's more like how refs work in independent repos.

Reply:
> Right, if we use file: links then we don't run lerna bootstrap at all. 

Now I don't recall if that statement is correct. 

I think we may have to run `npm ci` and then `lerna bootstrap` but like you discovered you cannot do that as part of an npm postinstall script like you might expect.